### PR TITLE
[KYUUBI #3818] [Subtask] [PySpark] set local properties to spark context for python scripts

### LINKED ISSUE #3818: [Subtask] [PySpark] set kyuubi.session.user in withLocalProperties #3817
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

- add `withLocalProperties` in `ExecutePython`
- set Spark local properties `kyuubi.session.user`, `kyuubi.statement.id`, `spark.scheduler.pool`, `spark.scheduler.pool` in withLocalProperties for other feature relied on it , e.g. Authz plugin.

### Parent issue

#3780

### Are you willing to submit PR?

- [X] Yes. I can submit a PR independently to improve.
- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

to close #3818 .
- add `withLocalProperties` in `ExecutePython`
- set Spark local properties `kyuubi.session.user`, `kyuubi.statement.id`, `spark.scheduler.pool`, `spark.scheduler.pool` in withLocalProperties for other feature relied on it , e.g. Authz plugin.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
cc @zhouyifan279

**Comment by bowenliang123** [comment]
cc @cfmcgrady

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Comment by cfmcgrady** [comment]
To be clear, in the current python-mode implementation, all the PySpark sessions are bound to the same spark session.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala [hunk 1]
```diff
@@ -63,6 +64,19 @@ class ExecutePython(
       new ArrayFetchIterator[Row](Array(Row(output, status, ename, evalue, Row(traceback: _*))))
   }
 
+  override protected def withLocalProperties[T](f: => T): T = {
+    val setSparkLocalProperties = (key: String, value: String) => {
+      worker.runCode(s"spark.sparkContext.setLocalProperty('$key', '$value')")
```
Comment (cfmcgrady) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala:69]: we should also set other properties like what we did in scala-mode.

https://github.com/apache/incubator-kyuubi/blob/dcf0ce6dcf23f3863f805e8039deda3ab66664c8/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala#L76-L94

**Comment by bowenliang123** [comment]
> To be clear, in the current python-mode implementation, all the PySpark sessions are bound to the same spark session.

Yes, well noticed. At least, Kyuubi session and pyspark process are separated.
This PR is intended to align local properties with SQLOperation.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala (hunk 1 above)
Comment (bowenliang123) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecutePython.scala:69]: OK. Not sure whether they are used in Pyspark.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**